snap/validate.go

```diff
@@ -351,6 +351,88 @@ func validateProvenance(prov string) error {
 	return naming.ValidateProvenance(prov)
 }
 
+type gpioChipLinesOverlapError struct {
+	chip, slotA, slotB string
+	spanA, spanB       strutil.RangeSpan
+}
+
+func (e *gpioChipLinesOverlapError) Error() string {
+	return fmt.Sprintf(`invalid "lines" attribute: chip %q has overlapping line spans: %q in slot %q overlaps with %q in slot %q`,
+		e.chip, e.spanB.String(), e.slotB, e.spanA.String(), e.slotA,
+	)
+}
+
+func validateGpioChardevSlots(info *Info) error {
+	// XXX: should this be relaxed to validate all types in case
+	// gpio-chardev slots are allowed for snap apps later?
```

Comment:
that's very unlikely, the only hw interface we let define not on gadget is custom-device